eis_toolkit/raster_processing/unique_combinations.py

```diff
@@ -0,0 +1,68 @@
+from typing import List, Dict
+
+import itertools
+import numpy as np
+import rasterio
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+
+
+def _unique_combinations(  # type: ignore[no-any-unimported]
+    list_of_rasters: List[rasterio.io.DatasetReader],
+    path_to_output_file: str
+) -> rasterio.io.DatasetReader:
+
+    bands = []
+    for raster in list_of_rasters:
+        bands.append(raster.read())
+
+    output_band = np.empty_like(bands[0])
+    combinations: Dict[tuple, int] = {}
+    for row, column in itertools.product(range(len(bands[0])), range(len(bands[0][0]))):
+
+        combination = tuple(bands[band][row][column] for band, _ in enumerate(bands))
+
+        if combination not in combinations:
+            combinations[combination] = len(combinations) + 1
+        output_band[row][column] = combinations[combination]
+
+    with rasterio.open(path_to_output_file, 'w', **list_of_rasters[0].meta) as write_raster:
+        write_raster.write(output_band, 1)
+    output_raster = rasterio.open(path_to_output_file)
+    return output_raster
```

Comment:
All the other raster processing functions so far have given "out_image, out_meta" (data array, meta) as the output, and we could follow that convention here too. Writing will be done in a separate function.

Reply:
Thank you for the info. I was unaware of this, albeit the fact that every other function gives out those outputs. Maybe this is something that should be mentioned to @jtorppa, too.